src/w3d/renderer/rinfo.cpp

```diff
@@ -0,0 +1,104 @@
+/**
+ * @file
+ *
+ * @author tomsons26
+ *
+ * @brief
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "rinfo.h"
+#include "refcount.h"
+#include <captainslog.h>
+
+RenderInfoClass::RenderInfoClass(CameraClass &camera) :
+    m_camera(camera),
+    field_4(0),
+    field_8(0),
+    field_C(0),
+    m_alphaOverride(1.0f),
+    m_opacityOverride(1.0f),
+    m_emissiveScale(1.0f),
+    m_lightEnvironment(nullptr),
+    field_20(0),
+    m_additionalMaterialPassCount(0),
+    field_A8(0),
+    m_overrideFlagLevel(0)
+{
+    for (int i = 0; i < ADDITIONAL_MATERIAL_PASSES; ++i) {
+        m_additionalMaterialPassArray[i] = nullptr;
+    }
+
+    for (int i = 0; i < MAX_OVERRIDE_FLAG_LEVEL; ++i) {
+        m_overrideFlag[i] = RINFO_OVERRIDE_DEFAULT;
+    }
+}
+
+inline RenderInfoClass::~RenderInfoClass()
+{
+    captainslog_assert(!m_additionalMaterialPassCount);
+
+    while (m_additionalMaterialPassCount) {
+        Pop_Material_Pass();
+    }
+}
+
+void RenderInfoClass::Push_Material_Pass(MaterialPassClass *matpass)
+{
+    if (m_additionalMaterialPassCount <= ADDITIONAL_MATERIAL_PASSES) {
+        if (matpass != nullptr) {
+            // TODO temp until MaterialPassClass is in
+            ((RefCountClass *)matpass)->Add_Ref();
+        }
+        m_additionalMaterialPassArray[m_additionalMaterialPassCount++] = matpass;
+    } else {
+        ++field_A8;
```

Comment:
Is this used anywhere else? It looks to me like is a counter for additional passes that are requested, but there is no room for, perhaps a debug counter for logging if additional passes are being requested?

Reply:
Haven't been able to find